subsys/bluetooth/controller/ll_sw/nordic/lll/lll_clock.c

```diff
@@ -82,43 +96,26 @@ int lll_hfclock_on(void)
 
 int lll_hfclock_on_wait(void)
 {
-	struct k_sem sem_clock_wait;
-	struct clock_control_async_data async_data = {
-		.cb = clock_ready,
-		.user_data = &sem_clock_wait,
-	};
-	int err;
-
-	k_sem_init(&sem_clock_wait, 0, 1);
-
-	err = clock_control_async_on(dev, CLOCK_CONTROL_NRF_SUBSYS_HF,
-				     &async_data);
-	LL_ASSERT(!err);
-
-	err = k_sem_take(&sem_clock_wait, K_MSEC(HFCLOCK_TIMEOUT_MS));
-
-	DEBUG_RADIO_XTAL(1);
-
-	return err;
+	return on_wait(&hf_state, HFCLOCK_TIMEOUT_MS);
 }
 
 int lll_hfclock_off(void)
 {
 	int err;
+	struct onoff_manager *hf_mgr =
+		z_nrf_clock_control_get_onoff(CLOCK_CONTROL_NRF_SUBSYS_HF);
 
-	/* turn off radio clock in non-blocking mode. */
-	err = clock_control_off(dev, CLOCK_CONTROL_NRF_SUBSYS_HF);
-	if (!err) {
-		DEBUG_RADIO_XTAL(0);
-	} else if (err == -EBUSY) {
-		DEBUG_RADIO_XTAL(1);
+	if (hf_refcnt < 1) {
+		return -EAGAIN;
 	}
 
-	return err;
-}
+	if (atomic_dec(&hf_refcnt) > 1) {
```

Comment:
```suggestion
	if (atomic_dec(&hf_refcnt) > 0) {
```

Reply:
`atomic_dec` returns previous value. So 1 means that refcnt reached 0.